.travis.yml

```diff
@@ -57,51 +67,68 @@ matrix:
     - &mac
       os: osx
       osx_image: xcode9.2
-      if: branch != master
+      if: type != pull_request
       env: MACOSX_DEPLOYMENT_TARGET=10.7 TARGET=x86_64-apple-darwin NO_ADD=1
     - <<: *mac
+      if: branch != master
       env: MACOSX_DEPLOYMENT_TARGET=10.7 TARGET=i686-apple-darwin
 
     - name: rustup-init.sh on CentOS 6
       language: minimal
       install:
       script:
         - docker run
-            --volume "$TRAVIS_BUILD_DIR":/checkout:ro
-            --workdir /checkout
-            --rm
-            -it
-            centos:6
-            sh ./ci/raw_init.sh
+          --volume "$TRAVIS_BUILD_DIR":/checkout:ro
+          --workdir /checkout
+          --rm
+          -it
+          centos:6
+          sh ./ci/raw_init.sh
       before_deploy:
       deploy:
 
+    - &windows
+      os: windows
+      env: TARGET=x86_64-pc-windows-msvc NO_ADD=1 EXE_EXT=.exe
+      services: nil
+      language: bash
+      before_install:
+        - choco install strawberryperl
+        - export PATH="/c/Strawberry/perl/bin:$PATH"
+
```

Comment:
Also we might want to install sccache on Windows as well. But let's do it
as follow up PR.

Reply:
Yeah, sccache is definitely a some-future time, though I am talking to Travis today about how to improve things for Rust people on Travis, maybe I'll suggest they offer an sccache addon.

Reply:
I've spoken with them and they're going to have a think, they may even contact you on Telegram.